parity/signer.rs

```diff
@@ -48,13 +49,18 @@ pub fn codes_path(path: &Path) -> PathBuf {
 	p
 }
 
-pub fn execute(ws_conf: rpc::WsConfiguration, ui_conf: rpc::UiConfiguration) -> Result<String, String> {
-	Ok(generate_token_and_url(&ws_conf, &ui_conf)?.message)
+pub fn execute(ws_conf: rpc::WsConfiguration, ui_conf: rpc::UiConfiguration, logger_config: LogConfig) -> Result<String, String> {
+	Ok(generate_token_and_url(&ws_conf, &ui_conf, &logger_config)?.message)
 }
 
-pub fn generate_token_and_url(ws_conf: &rpc::WsConfiguration, ui_conf: &rpc::UiConfiguration) -> Result<NewToken, String> {
+pub fn generate_token_and_url(ws_conf: &rpc::WsConfiguration, ui_conf: &rpc::UiConfiguration, logger_config: &LogConfig) -> Result<NewToken, String> {
 	let code = generate_new_token(&ws_conf.signer_path).map_err(|err| format!("Error generating token: {:?}", err))?;
 	let auth_url = format!("http://{}:{}/#/auth?token={}", ui_conf.interface, ui_conf.port, code);
+	let paint = |c: Style, t: String| match &logger_config.color {
```

Comment:
given that `paint` is only used once, the call to it could be replaced with the body. also, you can `match logger_config.color` directly, without borrowing it (bools are always copied).